cmd/demo/smartContracts/decode/decode.go

```diff
@@ -0,0 +1,214 @@
+package main
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/klever-io/klever-go-sdk/cmd/demo"
+	"github.com/klever-io/klever-go-sdk/provider"
+)
+
+type HexOutput struct {
+	Data struct {
+		Data string `json:"data"`
+	} `json:"data"`
+	Error string `json:"error"`
+	Code  string `json:"code"`
+}
+
+type ReturnData struct {
+	ReturnData []string `json:"returnData"`
+	ReturnCode string   `json:"returnCode"`
+}
+
+type QueryOutput struct {
+	Data struct {
+		Data ReturnData `json:"data"`
+	} `json:"data"`
+}
+
+func scReq(endpoint, funcName, scAddress string, args ...string) ([]byte, error) {
+	jsonArgs, err := json.Marshal(args)
+	if err != nil {
+		return nil, fmt.Errorf("error marshaling arguments: %v", err)
+	}
+
+	var argsString string
+	if len(args) == 0 {
+		argsString = `[]`
+	} else {
+		argsString = string(jsonArgs)
+	}
+
+	bodyTemplate := `{
+		"scAddress":"%s",
+		"funcName":"%s",
+		"args":%s
+	}`
+
+	bodyStr := fmt.Sprintf(bodyTemplate, scAddress, funcName, argsString)
+
+	bodyReader := strings.NewReader(bodyStr)
+
+	req, err := http.NewRequest(
+		"POST",
+		fmt.Sprintf("https://node.testnet.klever.finance/vm/%s", endpoint),
+		bodyReader,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error creating http request: %w", err)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("error executing http request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading http response body: %w", err)
+	}
+
+	return body, nil
+}
+
+func parseHex(hexBytes []byte, abiPath, funcName string, kc provider.KleverChain) (interface{}, error) {
+	var h HexOutput
+
+	if err := json.Unmarshal(hexBytes, &h); err != nil {
+		return nil, err
+	}
+
+	if h.Error != "" {
+		return nil, fmt.Errorf("error requesting kleverchain vm: %s", h.Error)
+	}
+
+	parser := kc.NewScOutputParser()
+
+	abi, err := os.Open(abiPath)
+	if err != nil {
+		return nil, err
+	}
+
+	err = parser.LoadAbi(abi)
+	if err != nil {
+		return nil, err
+	}
+
+	parsedValue, err := parser.DecodeHex(funcName, h.Data.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	return parsedValue, nil
+}
+
+func main() {
+	_, _, kc, err := demo.InitWallets()
+	if err != nil {
+		panic(err)
+	}
+
+	const ScAddress string = "klv1qqqqqqqqqqqqqpgqz0ce8rdktkap33cnmup545543pxy4f67d20qpkcuus"
+
+	var wg sync.WaitGroup
+	wg.Add(3)
+
+	go func() {
```

Comment:
does the operations need to be done in a goroutine? rather create a function and break down the execution...

Reply:
it is really not necessary to use goroutines. i used it because this demo makes 3 different http requests, doing this synchronously would make it take longer to complete.

but [here](https://github.com/klever-io/klever-go-sdk/pull/18/commits/9fc8e6f2712169328f3912c56718d94af1a464da) i adjusted it to your suggestion